### PR TITLE
Add "raw_path" to TestClient request scope

### PR DESCRIPTION
**Pull request by trimailov**
`raw_path` was missing in `TestClient` request's scope, whereas `uvicorn` has `raw_path` in request's scope. This problem makes it impossible to write tests, when parsing raw path is necessary.

I've created this app to prove this:
```
> cat example.py
from starlette.applications import Starlette
from starlette.responses import PlainTextResponse
from starlette.routing import Route
from starlette.testclient import TestClient


async def value(request):
    value = request.path_params['value']
    path = request.scope['path']
    raw_path = request.scope['raw_path']
    return PlainTextResponse(
        f'value: {value}\npath: {path}\nraw_path: {raw_path}\n'
    )


app = Starlette(debug=True, routes=[
    Route('/{value:path}', value),
])


def test_app():
    client = TestClient(app)
    response = client.get('/hel%2Flo')  # /hel/lo
    result = "value: hel/lo\npath: /hel/lo\nraw_path: /hel%2Flo\n"
    assert response.text == result
```

Launch the app with `uvicorn`:

```
> env/bin/uvicorn example:app
```

Test it with `curl`:

```
> curl localhost:8000/hel%2flo
value: hel/lo
path: /hel/lo
raw_path: b'/hel%2flo'
```

Test app with `TestClient`:

```
> env/bin/python -c "import example; example.test_app()"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/justas/work/tilaajavastuu/starlette_test/example.py", line 28, in test_app
    response = client.get('/hel%2Flo')  # /hel/lo
<...>
  File "/Users/justas/work/tilaajavastuu/starlette_test/example.py", line 14, in value
    raw_path = request.scope['raw_path']
KeyError: 'raw_path'
```

### CODE REVIEW

**Comment by trimailov** [comment]
Apparently similar PR was made in #635.

File: starlette/testclient.py [hunk 1]
```diff
@@ -134,6 +134,7 @@ def send(
             scope = {
                 "type": "websocket",
                 "path": unquote(path),
+                "raw_path": path,
```
Comment (lovelydinosaur) [inline starlette/testclient.py]: The ASGI spec mandates this as bytes, so `path.encode()` I guess?

Comment (trimailov) [reply starlette/testclient.py]: You are correct. Fixed it.

**Comment by Kludex** [comment]
@trimailov Thanks for the PR! :tada: 

Sorry for taking so long! Would you mind rebasing it?

**Comment by Kludex** [comment]
Closed in favor of #1445.

Author's commit are preserved.

**Pull request closed without merging**